server/odc-common/src/main/java/com/oceanbase/odc/common/crypto/RsaBytesEncryptor.java

```diff
@@ -0,0 +1,175 @@
+/*
+ * Copyright (c) 2023 OceanBase.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.oceanbase.odc.common.crypto;
+
+import java.security.InvalidKeyException;
+import java.security.Key;
+import java.security.KeyFactory;
+import java.security.KeyPair;
+import java.security.KeyPairGenerator;
+import java.security.NoSuchAlgorithmException;
+import java.security.SecureRandom;
+import java.security.interfaces.RSAPrivateKey;
+import java.security.interfaces.RSAPublicKey;
+import java.security.spec.InvalidKeySpecException;
+import java.security.spec.PKCS8EncodedKeySpec;
+import java.security.spec.X509EncodedKeySpec;
+
+import javax.crypto.BadPaddingException;
+import javax.crypto.Cipher;
+import javax.crypto.IllegalBlockSizeException;
+import javax.crypto.NoSuchPaddingException;
+
+import org.apache.commons.lang3.Validate;
+
+import com.oceanbase.odc.common.lang.Pair;
+import com.oceanbase.odc.common.util.EncodeUtils;
+
+import lombok.NonNull;
+
+/**
+ * @author gaoda.xy
+ * @date 2023/8/22 19:30
+ */
+public class RsaBytesEncryptor implements BytesEncryptor {
+
+    private static final String ALGORITHM_NAME = "RSA";
+    private static final int DEFAULT_KEY_SIZE = 1024;
+
+    private final RsaEncryptorType type;
+    private final Cipher encryptor;
+    private final Cipher decryptor;
+
+    public RsaBytesEncryptor(@NonNull RsaEncryptorType type, String publicKeyBase64, String privateKeyBase64) {
+        this.type = type;
+        if (type == RsaEncryptorType.ENCRYPT_MODE) {
+            Validate.notBlank(publicKeyBase64);
```

Comment:
Validate.notBlank should contains readable message

Reply:
I got you!